### PR TITLE
signup

### PR DESCRIPTION
**Pull request by kylymbekSarybashuulu**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,11 +36,22 @@
 			<scope>runtime</scope>
 			<optional>true</optional>
 		</dependency>
-		<dependency>
-			<groupId>org.postgresql</groupId>
-			<artifactId>postgresql</artifactId>
-			<scope>runtime</scope>
-		</dependency>
+<!--		<dependency>-->
```
Comment (i-Coderr) [inline pom.xml:39]: Remove commented dependencies

File: src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import us.peaksoft.gadgetarium.dto.AuthenticationResponse;
+import us.peaksoft.gadgetarium.dto.RegisterRequest;
+import us.peaksoft.gadgetarium.service.AuthService;
+
+@RestController
+@RequestMapping("api/gadgetarium/auth")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java:13]: rewrite to **api/public**

File: src/main/java/us/peaksoft/gadgetarium/dto/RegisterRequest.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package us.peaksoft.gadgetarium.dto;
+
+
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/RegisterRequest.java:3]: Remove line

File: src/main/java/us/peaksoft/gadgetarium/entity/User.java
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/entity/User.java:1]: Return back user entity class

File: src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+@RequiredArgsConstructor
+public class AuthConfiguration {
+    private final UserRepository userRepository;
+
+    @Bean
+    public UserDetailsService userDetailsService(){
+        return username -> userRepository.findByEmail(username).
+                orElseThrow(() -> new UsernameNotFoundException("User is not found"));
+    }
+    @Bean
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java:27]: Write to the next line

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfig {
+   // private final JwtAuthFilter jwtAuthFilter;
+    private final AuthenticationProvider authenticationProvider;
+    @Bean
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:19]: Write to the next line

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    public SecurityFilterChain securityFilterChain (HttpSecurity http) throws Exception{
+        http
+                .csrf()
+                .disable()
+                .authorizeHttpRequests()
+                .requestMatchers("/api/gadgetarium/auth/**").permitAll()
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:25]: api/public

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                .requestMatchers("/api/gadgetarium/auth/register")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:26]: Pls remove

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -5,3 +5,4 @@ spring.jpa.hibernate.ddl-auto=create
 spring.jpa.database-platform=org.hibernate.dialect.PostgreSQL9Dialect
 
 
+
```
Comment (i-Coderr) [inline src/main/resources/application.properties:8]: Pls remove

**Review by i-Coderr — COMMENTED** [review]
Commented at 11:35 pm 20/03

**Review by AzamatMuratbekov — COMMENTED** [review]
viwed

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java (hunk 7 above)
Comment (kylymbekSarybashuulu) [reply src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:25]: did

Comment (kylymbekSarybashuulu) [reply src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:25]: did

**Review by i-Coderr — COMMENTED** [review]
There are still unresolved comments

File: src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java (hunk 2 above)
Comment (kylymbekSarybashuulu) [reply src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java:13]: kyldym

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java (hunk 5 above)
Comment (kylymbekSarybashuulu) [reply src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:18]: ochyrdum

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java (hunk 6 above)
Comment (kylymbekSarybashuulu) [reply src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:24]: kyldym

**Review by i-Coderr — APPROVED** [review]
Ok

**Pull request merged**